packages/babel-parser/src/parser/statement.ts

```diff
@@ -346,15 +346,28 @@ export default abstract class StatementParser extends ExpressionParser {
     );
   }
 
-  startsUsingForOf(): boolean {
-    const { type, containsEsc } = this.lookahead();
+  allowsForUsing(): boolean {
+    const { type, containsEsc, end } = this.lookahead();
     if (type === tt._of && !containsEsc) {
-      // `using of` must start a for-lhs-of statement
-      return false;
-    } else if (tokenIsIdentifier(type) && !this.hasFollowingLineBreak()) {
+      // `for( using of` must start either a for-lhs-of statement
+      // or a for lexical declaration
+      const nextCharAfterOf = this.input.charCodeAt(
+        this.nextTokenStartSince(end),
+      );
```

Comment:
We can modify `lookaheadCharCode` as follows and use it.
```
  lookaheadCharCode(pos = this.state.pos): number {
    return this.input.charCodeAt(this.nextTokenStartSince(pos));
  }
```

Reply:
Since it is a method in hot path, I avoided the default parameter and introduced [lookaheadCharCodeSince](https://github.com/babel/babel/pull/17254/commits/e9ea20f5040a31db76892b6011c8dd005eade4d8).